### ISSUE TITLE
Return Array in Transaction

### ISSUE THREAD

**Issue by amirrezamahyari** [orientechnologies/orientjs#415]
Hi

when i run this query i get error :
```
 await db.let('q1', db.select().from('Person').where({ id: 49595 }))
	.let('q2', db.select().from('Product').where({ id: 1245 }))
        .commit()
	.return('[$q1,$q2]')
	.all();
```

 error in terminal:

```

(node:30335) UnhandledPromiseRejectionWarning: OrientDB.ConnectionError [1]: Remote server closed the connection.
    at ONetworkConnection.handleSocketEnd (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/network/conn.js:389:9)
    at Socket.emit (events.js:215:7)
    at endReadableNT (_stream_readable.js:1199:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
From previous event:
    at ONetworkConnection._sendOp (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/network/conn.js:80:10)
    at ONetworkConnection.send (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/network/conn.js:65:17)
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:174:27
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:187:18
From previous event:
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:185:10
From previous event:
    at SessionManager.withSession (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:180:12)
From previous event:
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:218:16
    at processImmediate (internal/timers.js:439:21)
From previous event:
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:210:15
From previous event:
    at SessionManager.withRetrySession (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:207:12)
    at SessionManager.send (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/session.js:155:17)
    at ODatabaseSession.send (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database.js:51:32)
    at ODatabaseSession._internalExec (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database.js:285:24)
    at ODatabaseSession.execStream (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database.js:111:12)
    at ODatabaseSession.execute (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database.js:173:17)
    at ODatabaseSession.batch (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database.js:155:17)
    at /Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database-query.js:34:15
From previous event:
    at SessionQuery.exec (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/client/database/database-query.js:29:12)
    at SessionQuery.all (/Users/amiza/Documents/Projects/Movie/server/node_modules/orientjs/lib/db/query.js:70:17)
```

and error in odb console is : 

```
$ANSI{green {db=movie_db}} Error while binary response serialization
java.lang.UnsupportedOperationException: LINKBAG should never appear in a projection
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serializeValue(OResultSerializerNetwork.java:428)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serialize(OResultSerializerNetwork.java:132)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serializeValue(OResultSerializerNetwork.java:394)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.writeEmbeddedCollection(OResultSerializerNetwork.java:567)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serializeValue(OResultSerializerNetwork.java:401)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.writeEmbeddedCollection(OResultSerializerNetwork.java:567)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serializeValue(OResultSerializerNetwork.java:401)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.serialize(OResultSerializerNetwork.java:132)
	at com.orientechnologies.orient.core.serialization.serializer.result.binary.OResultSerializerNetwork.toStream(OResultSerializerNetwork.java:651)
	at com.orientechnologies.orient.client.remote.message.OMessageHelper.writeProjection(OMessageHelper.java:411)
	at com.orientechnologies.orient.client.remote.message.OMessageHelper.writeResult(OMessageHelper.java:448)
	at com.orientechnologies.orient.client.remote.message.OQueryResponse.write(OQueryResponse.java:56)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sessionRequest(ONetworkProtocolBinary.java:343)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:212)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
```

**when i  use .return('$q1') or .return('$q2')  everything is fine but in array i get error
i use pool connection ( i think Previously when i use legacy version , didnt get error and i got results)**

**orientjs: 3.0.8
orientdb:3.0.24**

**Comment by amirrezamahyari** [comment]
same issue with Batch Scripts

```
await db.batch(
    `BEGIN;
     LET q1 = SELECT * FROM Person WHERE id = 49595;
     LET q2 = SELECT * FROM Product WHERE id = 1245;
    COMMIT ;
    RETURN [$q1,$q2] `
    )
   .all();
```

**Comment by amirrezamahyari** [comment]
ok , i tested on legacy and now i can confirm that works on it

**Comment by wolf4ood** [comment]
Hi @amirrezamahyari 

ok thanks. Most probably it's an issue on the server with the new protocol.
Let me check

**Comment by luigidellaquila** [comment]
Hi @amirrezamahyari 

I just pushed a fix that solves the problem. The fix will be released with v 3.0.27

Thanks

Luigi
